src/Components/EnvironmentPicker/EnvironmentPicker.tsx

```diff
@@ -133,150 +141,131 @@ const EnvironmentPicker = ({
 
     // set initial values based on props and local storage
     useEffect(() => {
-        let environmentUrls: string[] = [];
-        if (environmentUrl) {
-            // passed environmentUrl has precedence over the selected environment in localstorage, if enabled
-            environmentUrls = [environmentUrl];
+        // START of initializing ADT Instance url options
+        let adtInstances: Array<string> = [];
+        let adtInstanceToEdit: string;
+        if (adtInstanceUrl) {
+            adtInstanceToEdit = adtInstanceUrl; // passed prop has precedence over the localstorage, if enabled
+            adtInstances = [adtInstanceUrl];
         }
         if (isLocalStorageEnabled) {
-            const environmentUrlsInLocalStorage: Array<string> = getEnvironmentUrlsFromLocalStorage(
+            const selectedAdtInstanceInLocalStorage = getSelectedAdtInstanceFromLocalStorage(
                 localStorageKey
             );
-
-            if (
-                environmentUrl &&
-                !environmentUrlsInLocalStorage.includes(environmentUrl)
-            ) {
-                environmentUrlsInLocalStorage.push(environmentUrl);
-            }
-
-            environmentUrls = environmentUrls.concat(
-                environmentUrlsInLocalStorage.filter(
-                    (item) => environmentUrls.indexOf(item) < 0
+            adtInstanceToEdit =
+                adtInstanceToEdit ?? selectedAdtInstanceInLocalStorage?.url;
+
+            const adtInstancesInLocalStorage =
+                getAdtInstanceOptionsFromLocalStorage(localStorageKey) || [];
+            const adtInstanceUrlsInLocalStorage: Array<string> =
+                adtInstancesInLocalStorage?.map((a) => a.url) || [];
+            adtInstances = adtInstances.concat(
+                adtInstanceUrlsInLocalStorage.filter(
+                    (item) => adtInstances.indexOf(item) === -1
                 )
             );
         }
         environmentPickerDispatch({
-            type: EnvironmentPickerActionType.SET_ENVIRONMENT_ITEMS,
+            type: EnvironmentPickerActionType.SET_ADT_INSTANCE_ITEMS,
             payload: {
-                environmentItems: {
-                    environments: environmentUrls,
-                    environmentToEdit: environmentUrl
+                adtInstanceItems: {
+                    adtInstances,
+                    adtInstanceToEdit
                 }
             }
         });
+        // END of initializing ADT Instance url options
 
-        let containerUrls: Array<string> = [];
+        // START of initializing Storage account url options
+        let storageAccounts: Array<string> = [];
+        let storageAccountToEdit: string;
         if (storage?.containerUrl) {
-            containerUrls = [storage.containerUrl];
+            storageAccountToEdit = getStorageAccountUrlFromContainerUrl(
+                storage?.containerUrl
+            ); // passed prop has precedence over the localstorage, if enabled
```

Comment:
If this is the case, shouldn't the local storage if be an else if?

Reply:
the local storage check is couple of lines down (#190): so if not already initialized the value by passed prop, set it to local storage value
```
const selectedStorageAccountInLocalStorage = getSelectedStorageAccountFromLocalStorage();
            storageAccountToEdit =
                storageAccountToEdit ??
                selectedStorageAccountInLocalStorage?.url;
```